src/cider/nrepl/middleware/info.clj

```diff
@@ -308,12 +308,12 @@
 
 (defn eldoc-reply
   [msg]
-  (let [info (info msg)]
-    (if-let [arglists (extract-arglists info)]
-      (merge (extract-ns-or-class info)
-             (extract-name-or-member info)
-             {:eldoc (format-arglists arglists)})
-      {:status :no-eldoc})))
+  (if-let [info (info msg)]
+    (merge (extract-ns-or-class info)
+           (extract-name-or-member info)
+           {:doc (:doc info)}
+           {:eldoc (-> info extract-arglists format-arglists)})
```

Comment:
Btw, we might always return to the docstring and some `:type` to the response as well, so it would be easier to check what exactly did we end up resolving in the UI. The extra dostring for functions might also be displayed somehow in the eldoc if we figure out a creative way to do so.